tensorboard/components/tf_scalar_dashboard/tf-scalar-dashboard.html

```diff
@@ -100,165 +96,196 @@
+          computed: '_computeSmoothingEnabled(_smoothingWeight)',
         },
         _ignoreYOutliers: {
           type: Boolean,
           value: storage.getBooleanInitializer('_ignoreYOutliers', true, true),
           observer: '_ignoreYOutliersObserver',
         },
         _xType: {
-          type: String,
-          value: "step"
+          type: String,  // ["step", "relative", "wall"]
+          value: "step",
         },
-        _runsColorScale: {
+        _requestManager: {
           type: Object,
-          value: function() {
-            return {
-              scale: runsColorScale,
-            };
-          },
+          value: () => new RequestManager(50),
         },
       },
-      attached: function() {
-        this.async(function() {
-          this.fire("rendered");
-        });
-      },
       _showDownloadLinksObserver: storage.getBooleanObserver(
           '_showDownloadLinks', /*default=*/ false, /*useLocalStorage=*/ true),
       _smoothingWeightObserver: storage.getNumberObserver(
           '_smoothingWeight', 0.6),
       _ignoreYOutliersObserver: storage.getBooleanObserver(
           '_ignoreYOutliers', /*default=*/ true, /*useLocalStorage=*/true),
-      _computeSmoothingEnabled: function(_smoothingWeight) {
+      _computeSmoothingEnabled(_smoothingWeight) {
         return _smoothingWeight > 0;
       },
-      toggleLogScale: function(e) {
-        var currentTarget = Polymer.dom(e.currentTarget);
-        var button = currentTarget.parentNode.querySelector('.log-button');
-        var chart = currentTarget.parentNode.querySelector('vz-line-chart');
 
-        button.classList.toggle("log-button-selected");
-        chart.yScaleType = chart.yScaleType === 'log' ? 'linear' : 'log';
-        chart.redraw();
+
+      // DESTINATION: Similar code to be written in each plugin.
+      ready() {
+        this.reload();
+      },
+      reload() {
+        this._fetchTags().then(() => {
+          this._reloadCharts();
+        });
+      },
+      _fetchTags() {
+        const url = getRouter().pluginRoute('scalars', '/tags');
+        return this._requestManager.request(url).then(runToTag => {
+          if (_.isEqual(runToTag, this.runToTag)) {
+            // No need to update anything if there are no changes.
+            return;
+          }
+          const runs = getRuns(runToTag);
+          const tags = getTags(runToTag);
+          this.set("runs", runs);
+          this.set("tags", tags);
+          this.set("dataNotFound", tags.length === 0);
+          this.set("runToTag", runToTag);
+        });
+      },
+      _reloadCharts() {
+        this.querySelectorAll('tf-scalar-chart').forEach(chart => {
+          chart.reload();
+        });
+      },
+
+      // DESTINATION: Unknown. Pretty sure this does nothing, but some
+      // test code tries to listen to it: e2eTests.ts, fastTabSwitch.ts.
+      attached() {
+        this.async(() => {
+          this.fire("rendered");
```

Comment:
Hmm, okay. It looks like one test, `e2eTests.ts`, just checks that these indeed are fired, while the only other one [1], `fastTabSwitch.ts`, seems like something that we could do without… :-/

(I ask because I don't like polluting the public API with this.)

[1]: according to `on\(.render f:tensorboard`, and also a previous cursory review of `"rendered" f:tensorboard`.